CodeEdit/Features/Git/SourceControlManager.swift

```diff
@@ -43,14 +48,26 @@ final class SourceControlManager: ObservableObject {
     }
 
     /// Refresh all changed files and refresh status in file manager
-    func refresAllChangesFiles() async {
+    func refreshAllChangedFiles() async {
         do {
-            var changedFiles: [CEWorkspaceFile] = []
+            var fileDictionary = [URL: CEWorkspaceFile]()
 
+            // Process changed files
             for item in try await gitClient.getChangedFiles() {
-                changedFiles.append(.init(url: item.fileLink, changeType: item.changeType))
+                fileDictionary[item.fileLink] = CEWorkspaceFile(
+                    url: item.fileLink,
+                    changeType: item.changeType,
+                    staged: false
+                )
             }
 
+            // Update staged status
+            for item in try await gitClient.getStagedFiles() {
+                fileDictionary[item.fileLink]?.staged = true
+            }
+
+            let changedFiles = Array(fileDictionary.values.sorted())
```

Comment:
Let's test the performance of this when there's 1k+ changed files. We can add a comment to get back to it later

Reply:
If we don't do this the changed files appear in a random order which changes each time the view is re-initialized.

Reply:
I was thinking that it might be more of an issue of constantly allocating a new array, especially if it is large, on every refresh (I believe this function gets called every time a file system event occurs). I think this is something that might need some profiling, and we can experiment with using something like a heap or some other structure in the future.

Reply:
We can add a todo to this to profile it and explore a different data structure if needed. For now this is okay, at worse I'd expect a few bytes for the file URL plus a couple for the workspace file per item in the array.

```suggestion
            // TODO:  Profile
            let changedFiles = Array(fileDictionary.values.sorted())
```